### ISSUE TITLE
v3.2.0 CMake Error on target "nextcloudsync" (target_link_libraries)

### ISSUE THREAD

**Issue by barbosaaob** [nextcloud/desktop#3080]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
CMake configure completion

### Actual behaviour
CMake Error at src/libsync/CMakeLists.txt:141 (target_link_libraries):
  The keyword signature for target_link_libraries has already been used with
  the target "nextcloudsync".  All uses of target_link_libraries with a
  target must be either all-keyword or all-plain.

  The uses of the keyword signature are here:

   * src/libsync/CMakeLists.txt:122 (target_link_libraries)
   * src/libsync/CMakeLists.txt:135 (target_link_libraries)

### Steps to reproduce
```
cmake -DCMAKE_PREFIX_PATH=/usr/local/lib/qt5/cmake \
      -DCMAKE_C_FLAGS="-I/usr/local/include/inotify \
                       -I/usr/local/include/qtkeychain \
                       -I/usr/local/include" \
      -DCMAKE_CXX_FLAGS="-I/usr/local/include/inotify \
                         -I/usr/local/include/qtkeychain \
                         -I/usr/local/include" \
      -DCMAKE_EXE_LINKER_FLAGS="-L/usr/local/lib/inotify \
                                -L/usr/local/lib \
                                -linotify \
                                -Wl,-rpath=/usr/local/lib/inotify" \
       -DBUILD_SHELL_INTEGRATION_NAUTILUS=OFF \
       -DBUILD_SHELL_INTEGRATION_DOLPHIN=OFF \
       -DBUILD_SHELL_INTEGRATION=OFF \
       -DUNIT_TESTING=ON \
       ..
```

### Client configuration
Client version: 3.2.0

Operating system: OpenBSD 6.9 (GENERIC.MP) #459: Fri Apr  9 11:31:33 MDT 2021

Qt version: 
qtbase-5.15.2p1
qtdeclarative-5.15.2
qtkeychain-0.10.0
qtlocation-5.15.2
qtquickcontrols2-5.15.2
qtsvg-5.15.2
qttools-5.15.2
qtwebchannel-5.15.2
qtwebengine-5.15.2p0
qtwebsockets-5.15.2

### Additional notes
I can build the client if I remove the PUBLIC tag on lines 123 and 135 on src/libsync/CMakeLists.txt
Is was introduced on 73a41d8e349bc63b0193722a30369ab23a48523b

**Comment by FlexW** [comment]
@barbosaaob Thanks for this report. I have no BSD system to test it but would it be possible that you test if [this pull request](https://github.com/nextcloud/desktop/pull/3084) resolves the issue.

**Comment by barbosaaob** [comment]
That solves the issue. Thank you.

**Comment by FlexW** [comment]
Should be fixed by #3059
